iothub/device/src/DirectMethod/DirectMethodRequest.cs

```diff
@@ -121,7 +121,7 @@ public bool TryGetPayload<T>(out T payload)
 
             try
             {
-                payload = PayloadConvention.GetObject<T>(Payload);
+                payload = DefaultPayloadConvention.Instance.GetObject<T>(Payload);
```

Comment:
This should be done with the user's selected payload convention. Presumably you have an e2e test with STJ selected using a custom class with only STJ attributes, right? It should fail this scenario.